xinference/core/supervisor.py

```diff
@@ -86,7 +90,10 @@ async def _choose_worker(self) -> xo.ActorRefType["WorkerActor"]:
         # TODO: better allocation strategy.
         min_running_model_count = None
         target_worker = None
-        for worker in self._worker_address_to_worker.values():
+        async with self._lock:
+            workers = list(self._worker_address_to_worker.values())
```

Comment:
Here, the lock is not required. All operations on the dictionary are synchronous, and any operation on the dictionary will not be interrupted by async calls.